File: weather-app/src/main/java/rcgonzalezf/org/weather/models/WeatherViewModel.java
package rcgonzalezf.org.weather.models;

import android.os.Parcelable;
import java.io.Serializable;

public interface WeatherViewModel extends Parcelable, Serializable {

  int getId();

  String getCityName();

  double getSpeed();

  String getHumidity();

  double getTemperature();

  String getDateTime();

  int getWeatherId();

  String getCountry();

  double getDeg();

  String getDescription();
}

File: weather-interface/src/main/java/org/rcgonzalezf/weather/common/models/WeatherViewModel.kt
package org.rcgonzalezf.weather.common.models

import android.os.Parcelable
import java.io.Serializable

interface WeatherViewModel : Parcelable, Serializable {
    val id: Int
    val cityName: String?
    val speed: Double
    val humidity: String?
    val temperature: Double
    val dateTime: String?
    val weatherId: Int
    val country: String?
    val deg: Double
    val description: String?
}
